src/patches/disable-node-cli.js

```diff
@@ -0,0 +1,15 @@
+export default async function disableNodeCli (compiler, next) {
+  if (compiler.options.enableNodeCli) {
+    return next()
+  }
+
+  const nodecc = await compiler.readFileAsync('src/node.cc')
+  const nodeccMarker = "argv[index][0] == '-'"
+
+  nodecc.contents = nodecc.contents.replace(
+    nodeccMarker,
+    nodeccMarker.replace('-', ']')
+  )
```

Comment:
Since the patterns are so common, I would create `compiler.replaceInFileAsync` and `compiler.setFileContentsAsync`.  Will make these modules super clear and concise.